src/compiler/checker.ts

```diff
@@ -11678,7 +11691,19 @@ namespace ts {
                 return undefined;
             }
             if (!propSet && !(checkFlags & CheckFlags.ReadPartial) && !indexTypes) {
-                return singleProp;
+                if (mergedInstantiations) {
+                    // No symbol from a union/intersection should have a `.parent` set (since unions/intersections don't act as symbol parents)
+                    // Unless that parent is "reconsituted" from the "first value declaration" on the symbol (which is likely different than its instantiated parent!)
+                    // They also have a `.containingType` set, which affects some services endpoints behavior
+                    const clone = createSymbolWithType(singleProp, (singleProp as TransientSymbol).type);

```

Comment:
The normal return path of this function sets CheckFlags.Deferred on the return value. Is that needed here too?

Reply:
Nope - the symbol, as a clone of another symbol, has a deferred type by default (lazy calculation of the .type field is the norm) and we don't need to override that type, like we do for the composited symbols.